src/Components/singleArticle.js

```diff
@@ -1,20 +1,57 @@
-import axios from 'axios'
 import {useParams, Link} from 'react-router-dom'
 import {useState, useEffect} from 'react'
-import { getArticleById } from '../Helpers/Api'
+import { getArticleById, changeArticleVotes } from '../Helpers/Api'
 
 const SingleArticle = () => {
     const [singleArticle, setSingleArticle] = useState({})
+    const [optimisticVotes, setOptimisticVotes] = useState(0)
+    const [successfulVote, setSuccessfulVote] = useState(false)
+    const [failedVote, setFailedVote] = useState(false)
 
     const {article_id} = useParams()
 
     useEffect(() => {
         getArticleById(article_id).then((data) => {
             setSingleArticle(data.data.article)
         })
-    }, [])
+    }, [article_id])
+
+    const incrementVote = () => {
+        setSuccessfulVote(false)
+        setFailedVote(false)
+        setOptimisticVotes((currentOptimisticVotes) => {
+            return currentOptimisticVotes + 1
+        })
+        changeArticleVotes(article_id, 1)
+        .then(()=> {
+            setSuccessfulVote(true)
+        })
+        .catch((err) => {
+            setOptimisticVotes((currentOptimisticVotes) => {
+                return currentOptimisticVotes - 1
+            })
+            setFailedVote(true)
+        })
+    }
+
+    const decrementVote = () => {
+        setSuccessfulVote(false)
+        setFailedVote(false)
+        setOptimisticVotes((currentOptimisticVotes) => {
+            return currentOptimisticVotes - 1
+        })
+        changeArticleVotes(article_id, -1)
+        .then(() => {
+            setSuccessfulVote(true)
+        })
+        .catch((err) => {
+            setOptimisticVotes((currentOptimisticVotes) => {
+                return currentOptimisticVotes + 1
+            })
+            setFailedVote(true)
+        })
+    }
```

Comment:
`incrementVote` and `decrementVote` look very similar. Might be best to refactor and make one reusable function you can use for both incrementing and decrementing the votes. Tip: make use of arguments!